griptape/mixins/rule_mixin.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import Any
+
+from attr import define, field
+
+from griptape.rules import Ruleset, Rule
+
+
+@define(slots=False)
+class RuleMixin:
+    DEFAULT_RULESET_NAME = "Default Ruleset"
+    ADDITIONAL_RULESET_NAME = "Additional Ruleset"
+
+    rulesets: list[Ruleset] = field(factory=list, kw_only=True)
+    rules: list[Rule] = field(factory=list, kw_only=True)
+    structure: Any = field(default=None, kw_only=True)
```

Comment:
`structure` needs to be defined in this class for Pyright to allow us to check for `self.structure` below. We can default to None, and if the class using the mixin has a structure, it will be overridden, allowing the Task to access its structure's rules (confirmed experimentally and via unit tests below)

Reply:
That raised instinctual circular import alarms to me but it checks out! Updated.